utils/bigclown/bigclown-gateway/files/init

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh /etc/rc.common
+
+START=98
+STOP=1
+
+USE_PROCD=1
+
+PROG=/usr/bin/bcg
+CONF=/tmp/etc/bigclown-gateway.conf
+
+append() {
+	local cfg="$1"
+	local uci_name="$2"
+	local out_name="$3"
+	local default="$4"
+	config_get val $cfg $uci_name $default
+	if [ -n "$val" ]; then
+	  echo "$out_name $val" >> $CONF
+	fi
+}
+
+start_service() {
+	config_load bigclown-gateway
+
+	rm -rf $CONF
+	echo "Generating bigclown-gateway config file in $CONF"
+
+	append gateway name 'name:' usb-dongle
+	# TODO add udev rules and use different default here
+	append gateway device 'device:' /dev/ttyUSB0
+	append gateway automatic_rename_kit_nodes 'automatic_rename_kit_nodes:' 1
+	append gateway base_topic_prefix 'base_topic_prefix:'
+
+	echo "mqtt:" >> $CONF
+	append mqtt host '  host:' localhost
+	append mqtt port '  port:' 1883
+	append mqtt cafile '  cafile:'
+	append mqtt certfile '  certfile:'
+	append mqtt keyfile '  keyfile:'
+
+	procd_open_instance
+	procd_set_param respawn ${respawn_threshold:-3600} ${respawn_timeout:-5} ${respawn_retry:-5}
+	procd_set_param command "$PROG" -c "$CONF" -W
+	procd_set_param stdout 1
+	procd_set_param stderr 1
+	procd_close_instance
+}
+
+stop_service() {
+	service_stop "$PROG"
+	ps | grep "$PROG"  | grep -v grep | sed -e 's/^\s*\([0-9]\+\)\s.*$/\1/' | xargs -r -- kill -9
```

Comment:
what on earth is your bigclown doing that procd isn't catching it?  also, pgrep is in the base install (even if pkill isn't) which, if this monstrosity _is_ necessary, would make it much simpler.  _most_ procd managed services can omit the stop_service function entirely.

Reply:
I suspect that it does not want to die on simple sigterm so they also send sigkill. I am going to drop that. It makes no sense. If it does not die the I am going to report that to upstream project because otherwise it is just stupid.